odoo/addons/base/models/res_lang.py

```diff
@@ -277,10 +277,13 @@ def write(self, vals):
         if 'code' in vals and any(code != vals['code'] for code in lang_codes):
             raise UserError(_("Language code cannot be modified."))
         if vals.get('active') == False:
+            inactive_users_lang = self.env['res.users'].search([('active', '=', False)]).mapped('lang')
             if self.env['res.users'].search_count([('lang', 'in', lang_codes)]):
                 raise UserError(_("Cannot deactivate a language that is currently used by users."))
             if self.env['res.partner'].search_count([('lang', 'in', lang_codes)]):
                 raise UserError(_("Cannot deactivate a language that is currently used by contacts."))
+            if any(code in inactive_users_lang for code in lang_codes):
```

Comment:
The following would be much more efficient:
```suggestion
            if self.env['res.users'].search_count([('lang', 'in', lang_codes), ('active', '=', False)]):
```
And by the way, it's almost the same query as above...